### ISSUE TITLE
Accessibility audit/support

### ISSUE THREAD

**Issue by benwiley4000** [benwiley4000/cassette#261]
We should try to make version 2 as accessible as possible. This means:
* Support for [text tracks](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/track) (may be somewhat involved to build an API around this)
* Support for screen readers... can a blind person read the UI easily? (only applicable to the default player implementation). Buttons should ideally have `aria-label` attributes explaining their function. Note that these should be swappable with other translations ideally.. (think about that for #247).
* Support for keyboard navigation for all functions - mostly need to make sure it's intuitive to adjust volume and progress via keyboard (only applicable to the default player implementation)

Resource: https://a11yproject.com/

**Comment by benwiley4000** [comment]
## For text tracks:
1. We should support a `textTracks` property on the playlist track object which is a list of objects describing the properties in an [`HTMLTrackElement`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLTrackElement). We will dynamically add tracks to the video element (this [article](https://iandevlin.com/blog/2015/02/javascript/dynamically-adding-text-tracks-to-html5-video/) has good advice).
2. We need a way to select which text tracks are active. Captions need to be able to be toggled on and off. For other `kind`s of text tracks (`subtitles`, `description`, `chapters`, `metadata`) I'm not sure how these are shown or not. How does the browser handle them? Do we need to do anything special since we're not displaying the browser's default video controls?
3. We need a way to select which language the user prefers. Which language is default? Does the developer need to hardcode this or can it be detected from the user's system?
4. We can have a boolean prop called `displayNativeCaptions` (or something) that is `true` by default but can be turned off to disable the browser's native text tracks display inside the video element.
5. We'll want to make the cues available via `playerContext`, which means making most or all of the data from the [VTTCue](https://w3c.github.io/webvtt/#the-vttcue-interface) interface available. We'll want to make all the active and possibly *all* the cues for the active track(s) available.

We should check out how related/similar libraries have handled APIs for text tracks. Most importantly we should see what options popular media libraries have for manipulating available text tracks from the user angle.

This might need to be spun out into a new issue since it might not all fit into the 2.0 release.

## For screen readers:
This can probably happen easily for all the buttons. For the ProgressBar stuff (volume and audio progress) it will require some research.

## For keyboard navigation:
Again this will require some thought but might be more immediately feasible than text track support. Do we need to look at using a native element like range selector? Probably best to figure this out before we go into beta because it could affect the API for ProgressBar.

**Comment by benwiley4000** [comment]
I forgot about #108. If we spin out text track support it can be tracked there.

**Comment by benwiley4000** [comment]
Keyboard navigation added in #395.

**Comment by benwiley4000** [comment]
Closing since all tasks are spun out into separate issues.
